### PR TITLE
Add meal planner screen

### PR DESCRIPTION
**Pull request by UtsavK55**
This PR consists of meal planner screen and changes related to it

### CODE REVIEW

File: src/screens/mealPlanner/MealPlanner.tsx [hunk 1]
```diff
@@ -15,12 +56,82 @@ const MealPlanner = () => {
...
+            {selectedDate.toLocaleDateString('en-US', {
+              month: 'long',
+              day: 'numeric',
+              year: 'numeric',
+            })}
+          </Text>
+          <Pressable style={styles.icon} onPress={onPressClear}>
+            <Text style={styles.delete}>Clear Plan</Text>
+          </Pressable>
+          {mealTypes.map(({mealId, mealName}) => {
```
Comment (02priyanshi) [inline src/screens/mealPlanner/MealPlanner.tsx:136]: extract to component

**Review by 02priyanshi — CHANGES_REQUESTED** [review]

File: src/constants/index.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 export const STORAGE_KEYS = {
   DARK_MODE: 'darkMode',
+  USER_DATA: 'userData',
+  FAVOURITE: 'FAVOURITE',
```
Comment (adiTrivsme) [inline src/constants/index.ts]: keep keys consitent

File: src/screens/mealDetails/MealDetails.tsx [hunk 3]
```diff
@@ -221,23 +248,132 @@ const MealDetails = () => {
...
+          value,
+        })) || [];
+      setMealPlan(mealPlanDetail);
+    }
+  };
+
+  const getFavorite = async () => {
+    const data = await getData(STORAGE_KEYS.FAVOURITE);
+    setFavorites(data || []);
+    const favorite = data.includes(recipeId);
```
Comment (adiTrivsme) [inline src/screens/mealDetails/MealDetails.tsx:277]: add blank line

File: src/screens/mealDetails/MealDetails.tsx [hunk 4]
```diff
@@ -221,23 +248,132 @@ const MealDetails = () => {
...
+    setIsFavorite(favorite);
   };
 
   useEffect(() => {
     getAllData();
+    getFavorite();
   }, []);
 
+  const mealName = mealTypes.find(item => item?.mealId == mealId)?.mealName;
```
Comment (adiTrivsme) [inline src/screens/mealDetails/MealDetails.tsx]: useMemo

File: src/screens/mealDetails/MealDetails.tsx [hunk 5]
```diff
@@ -221,23 +248,132 @@ const MealDetails = () => {
...
+
+  const {id, title, image} = recipeInfo;
+  const queryParams = {
```
Comment (adiTrivsme) [inline src/screens/mealDetails/MealDetails.tsx]: useMemo

File: src/screens/mealDetails/MealDetails.tsx [hunk 6]
```diff
@@ -77,40 +100,43 @@ const TabContent = ({activeTab, recipeInfo}: TabContentProps) => {
...
-      label: recipeInfo.dairyFree ? 'Non-Dairy' : 'Contains Dairy products',
-      value: '',
-    },
-    {
-      label: recipeInfo.glutenFree ? 'Gluten free' : 'Contains Gluten',
-      value: '',
-    },
-  ];
+  const recipeDetails = useMemo(
+    () => [
```
Comment (adiTrivsme) [inline src/screens/mealDetails/MealDetails.tsx]: this is not readable, you can define the separate function outside component for this all and call inside useMemo passing the `recipeInfo`

**Review by adiTrivsme — APPROVED** [review]

**Pull request merged**